src/minio/core/minio_client.py

```diff
@@ -81,3 +84,231 @@ async def test_connection(self) -> bool:
         except Exception as e:
             logger.error(f"Connection test failed: {e}")
             return False
+
+    async def create_bucket(self, bucket_name: str) -> bool:
+        """
+        Creates a new bucket on the MinIO server.
+
+        Args:
+            bucket_name: The name of the bucket to create.
+
+        Returns:
+            True if the bucket was created successfully.
+
+        Raises:
+            BucketOperationError: If the bucket creation fails for any reason.
+        """
+        try:
+            async with self._get_client() as client:
+                await client.create_bucket(Bucket=bucket_name)
+                logger.info(f"Created bucket: {bucket_name}")
+                return True
+        except Exception as e:
+            logger.error(f"Unexpected error creating bucket {bucket_name}: {e}")
+            raise BucketOperationError(f"Bucket creation failed: {e}") from e
+
+    async def bucket_exists(self, bucket_name: str) -> bool:
+        """
+        Checks if a bucket exists by sending a HEAD request.
+
+        NOTE: This method is designed to be used by minIO admin only where we have access to all buckets.
+
+        Args:
+            bucket_name: The name of the bucket to check.
+
+        Returns:
+            True if the bucket exists, False otherwise.
+
+        Raises:
+            BucketOperationError: If the check fails for reasons other than a 404 error.
+        """
+        try:
+            async with self._get_client() as client:
+                await client.head_bucket(Bucket=bucket_name)
+                return True
+        except ClientError as e:
+            # Reference: https://docs.aws.amazon.com/AmazonS3/latest/API/API_HeadBucket.html
+            if e.response["Error"]["Code"] == "404":
+                return False
+            else:
+                logger.error(f"Error checking bucket {bucket_name}: {e}")
+                raise BucketOperationError(f"Bucket check failed: {e}") from e
+        except Exception as e:
+            logger.error(f"Unexpected error checking bucket {bucket_name}: {e}")
+            raise BucketOperationError(f"Bucket check failed: {e}") from e
+
+    async def list_buckets(self) -> List[str]:
+        """
+        Retrieves a list of all bucket names from the MinIO server.
+
+        Returns:
+            A list of strings, where each string is a bucket name.
+
+        Raises:
+            BucketOperationError: If listing buckets fails.
+        """
+        try:
+            async with self._get_client() as client:
+                response = await client.list_buckets()
+                buckets = [bucket["Name"] for bucket in response.get("Buckets", [])]
+                logger.info(f"Listed {len(buckets)} buckets")
+                return buckets
```

Comment:
Technically there should probably be a limit / paginator here but that seems like an unlikely problem to need to solve for the reasonably near future

Reply:
yea - I think we would be fine for now.